tools/wake-format/emitter.cpp

```diff
@@ -214,7 +215,15 @@ auto Emitter::rhs_fmt() {
 }
 
 auto Emitter::pattern_fmt(cst_id_t stop_at) {
-  auto part_fmt = fmt().walk(is_expression, WALK_NODE).consume_wsnlc();
+  // clang-format off
+  auto part_fmt = fmt().match(
+   pred([](wcl::doc_builder&, ctx_t, CSTElement& node, const token_traits_map_t&){
+     return node.isNode();
+    }, fmt().walk(is_expression, WALK_NODE))
+   .otherwise(fmt().walk(WALK_TOKEN)))
+   .consume_wsnlc();
```

Comment:
The pattern format part should just be a single node so you don't have to do this. By splitting it up into the slash and non-slash part you can eliminate this entire thing.